File: framework/lib/border_radius.dart
part of uxml;

/**
 * This class represents immutable border radius for a control.
 *
 * @author ferhat@ (Ferhat Buyukkokten)
 */
class BorderRadius {
  /** Empty radius */
  static BorderRadius EMPTY;

  /** Sets or returns left top radius value */
  num topLeft;

  /** Sets or returns right top value */
  num topRight;

  /** Sets or returns bottom right value */
  num bottomRight;

  /** Sets or returns bottom left value */
  num bottomLeft;


  /**
   * Constructs an immutable border radius.
   */
  BorderRadius(this.topLeft, this.topRight, this.bottomRight, this.bottomLeft);

  /**
   * Constructs corner radius four equals radii.
   */
  BorderRadius.uniform(num size) {
    topLeft = size;
    topRight = size;
    bottomLeft = size;
    bottomRight = size;
  }

  /**
   * Checks equality.
   */
  bool equals(BorderRadius value) {
    return (value.topLeft == topLeft) &&
        (value.topRight == topRight) &&
        (value.bottomLeft == bottomLeft) &&
        (value.bottomRight == bottomRight);
  }

  /**
   * Returns true if all edges have same radius.
   */
  bool isUniform() {
    return (topLeft == topRight) && (bottomLeft == bottomRight) &&
        (topLeft == bottomLeft);
  }

  static void initialize() {
    EMPTY = new BorderRadius.uniform(0.0);
  }
}

File: plugin/com.hello.uxml.tools.ui/src/com/hello/uxml/tools/framework/BorderRadius.java
package com.hello.uxml.tools.framework;

/**
 * Represents immutable border radius of an element.
 *
 * @author ferhat
 *
 */
public class BorderRadius{

  private double topLeft;
  private double topRight;
  private double bottomRight;
  private double bottomLeft;

  /**
   * Empty margin object.
   */
  public static final BorderRadius EMPTY = new BorderRadius(0, 0, 0, 0);

  /**
   * Constructor.
   */
  public BorderRadius(double topLeft, double topRight, double bottomRight, double bottomLeft) {
    this.topLeft = topLeft;
    this.topRight = topRight;
    this.bottomRight = bottomRight;
    this.bottomLeft = bottomLeft;
  }

  public static BorderRadius create(double size) {
    return new BorderRadius(size, size, size, size);
  }

  /**
   * Returns top left radii.
   */
  public double getTopLeft() {
    return topLeft;
  }

  /**
   * Returns top right radii.
   */
  public double getTopRight() {
    return topRight;
  }

  /**
   * Returns bottom right radii.
   */
  public double getBottomRight() {
    return bottomRight;
  }

  /**
   * Returns bottom left radii.
   */
  public double getBottomLeft() {
    return bottomLeft;
  }

  @Override
  public String toString() {
    // Not using String.format due to GWT
    StringBuilder sb = new StringBuilder();
    sb.append(topLeft);
    sb.append(" ");
    sb.append(topRight);
    sb.append(" ");
    sb.append(bottomRight);
    sb.append(" ");
    sb.append(bottomLeft);
    return sb.toString();
  }
}
